docs/pages/machine-id/guides/circleci.mdx

```diff
@@ -0,0 +1,226 @@
+by which the Auth Server decides whether or not to allow a bot or node to join.
+
+Create a file named `tokenconfig.yaml`, ensuring that you replace
+`$ORGANIZATION_ID` and `$CONTEXT_ID` with the values from step 1.
+
+```yaml
+kind: token
+version: v2
+metadata:
+  name: circleci-demo
+  expires: "2100-01-01T00:00:00Z"
+spec:
+  roles: [Bot]
+  join_method: circleci
+  bot_name: circleci-demo
+  circleci:
+    organization_id: $ORGANIZATION_ID
+    allow:
+      - context_id: $CONTEXT_ID
+```
+
+Let's go over the token resource's fields in more detail:
+
+- `metadata.name` defines the name of the token. Note that this value will need
+to be used in other parts of the configuration later.
+- `metadata.expires` defines the date that the join token will expire. This
+example is set to the year `2100`.
+- `spec.bot_name` is the name of the Machine ID bot that this token will grant
+access to. Note that this value will need to be used in other parts of the
+configuration later.
+- `spec.roles` defines which roles that this token will grant access to. The
+value of `[Bot]` states that this token grants access to a Machine ID bot.
+- `spec.join_method` defines the join method the token is applicable for. Since
+this guide only focuses on CircleCI, you will set this to to `circleci`.
+- `spec.circleci.allow` is used to set rules for what CircleCI runs will be able
+to authenticate by using the token.
+
+Apply this to your Teleport cluster using `tctl`:
+
+```code
+$ tctl create -f tokenconfig.yaml
+```
+
+## Step 3/4. Create a Machine ID bot
+
+With the join token for the CircleCI project created, you now need to create a
+Machine ID bot that the token will grant access to. A Machine ID bot is a
+special type of Teleport user designed for access by machines, and can
+authenticate using a join token rather than forms of authentication more
+suitable to users (e.g such as SSO.)
+
+For this guide, we are using the default `access` role and explicitly stating
+that the bot should have access to the `ci-user` login on hosts. In production
+environments, we recommend creating a custom role for your CI/CD workflow and
+ensuring that this role has no more permissions than is needed for the workflow
+to complete its tasks.
+
+Use `tctl` to create the bot:
+
+```code
+$ tctl bots add circleci-demo --roles=access --logins=ci-user --token=circleci-demo
+```
+
+## Step 4/4. Configure CircleCI workflow
+
+With the token and bot now configured, you can now create a CircleCI workflow
+that can connect to your Teleport cluster.
+
+Our example workflow will connect to a SSH node named `my-node` and write a
+line to a file that links back to the CI/CD run. This is a trivial example and
+could be replaced with pushing new configuration or binaries over SSH.
+
+Open your Git repository and create a directory called `.circleci`. Then open
+a file called `config.yml` and insert the following configuration. Replace
```

Comment:
What would you think of:
- moving the part of the paragraph beginning at "Replace" after the snippet
- ending this paragraph at "configuration" with a colon to introduce the snippet

This way, once the user has a YAML file to edit, they can follow the instructions to replace different values.

Reply:
8c4f60c735